src/netlink/nl_l3.cc

```diff
@@ -831,9 +900,8 @@ int nl_l3::del_l3_neigh(struct rtnl_neigh *n) {
 
     // delete next hop
     rv = sw->l3_unicast_host_remove(ipv4_dst);
-  } else {
-    addr = rtnl_neigh_get_dst(n);
-
+  }
+  if (family == AF_INET6 && skip_addr_remove) {
```

Comment:
```suggestion
  } else if (family == AF_INET6 && skip_addr_remove) {
```

Also did you mean `!skip_addr_remove`?